cli/bpmetadata/repo_test.go

```diff
@@ -73,14 +74,74 @@ func TestGetBpSubmoduleName(t *testing.T) {
 			path: "testdata/bpmetadata/terraform-google-bp01/foo/submodule-01",
 			want: "",
 		},
+		{
+			name:      "simple - submodule with underscores",
+			path:      "testdata/bpmetadata/terraform-google-bp01/modules/submodule_01",
+			makeKebab: true,
+			want:      "submodule-01",
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got := getBpSubmoduleName(tt.path)
+			got := getBpSubmoduleName(tt.path, tt.makeKebab)
 			if got != tt.want {
 				t.Errorf("getBpSubmoduleName() = %v, want %v", got, tt.want)
 			}
 		})
 	}
 }
+
+func TestGetRepoDetailsFromRootBp(t *testing.T) {
+	tests := []struct {
+		name            string
+		path            string
+		wantRepoDetails repoDetail
+	}{
+		{
+			name: "root metadata does not exist",
+			path: "../testdata/bpmetadata/content/examples/simple_regional_beta/modules/submodule-01",
+			wantRepoDetails: repoDetail{
+				Source: &repoSource{
+					BlueprintRootPath: "../testdata/bpmetadata/content/examples/simple_regional_beta",
+				},
+			},
+		},
+		{
+			name: "root metadata exists but does not have source info",
+			path: "../testdata/bpmetadata/content/examples/acm/modules/submodule-01",
+			wantRepoDetails: repoDetail{
+				RepoName: "terraform-google-module",
+				Source: &repoSource{
+					BlueprintRootPath: "../testdata/bpmetadata/content/examples/acm",
+				},
+			},
+		},
+		{
+			name: "root metadata exists and has source info",
```

Comment:
That would be redundant here because this fn already takes in a blueprint path i.e. by the time this is executed, we already know where the blueprint is and this is just giving us the root path and other details for the blueprint, not the repo.